drivers/flash/Kconfig

```diff
@@ -128,6 +128,7 @@ config SOC_FLASH_NRF5
 	depends on FLASH && SOC_FAMILY_NRF5
 	default n
 	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_PAGE_LAYOUT if IS_MCUBOOT
```

Comment:
is this unique to nordic or should we always select FLASH_PAGE_LAYOUT for IS_MCUBOOT regardless of flash/SoC?

Meaning should we have:

select FLASH_PAGE_LAYOUT if FLASH_HAS_PAGE_LAYOUT in config IS_MCUBOOT (over in misc/Kconfig)

Reply:
Good point. @nvlsianpu thoughts?

Reply:
I agree witch this. If FLASH_PAGE_LAYOUT is supported by a target, it should be used by mcuboot. Support for targets without this might (rather will) be deprecated at some point.

Reply:
Addressed